test/client/fixtures/sandbox/node/document-test.js

```diff
@@ -494,22 +495,56 @@ if (!browserUtils.isIE) {
 }
 
 test('an iframe should not contain self-removing scripts after document.close (GH-871)', function () {
-    var iframe = document.createElement('iframe');
+    return createTestIframe()
+        .then(function (iframe) {
+            var iframeDocument = iframe.contentDocument;
 
-    iframe.id = 'test-gh-871';
+            iframeDocument.designMode = 'On';
+            iframeDocument.open();
+            iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
+            iframeDocument.close();
 
-    document.body.appendChild(iframe);
+            var selfRemovingScripts = nativeMethods.querySelectorAll.call(iframeDocument,
+                '.' + SHADOW_UI_CLASSNAME.selfRemovingScript);
+
+            strictEqual(selfRemovingScripts.length, 0);
+        });
+});
 
-    var iframeDocument = iframe.contentDocument;
+test('should not throw an error when document.defualtView is null', function () {
+    return new Promise(function (resolve, reject) {
+        var iframe         = document.createElement('iframe');
+        var loadEventCount = 0;
 
-    iframeDocument.designMode = 'on';
-    iframeDocument.open();
-    iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
-    iframeDocument.close();
+        iframe.id     = 'test' + Date.now();
+        iframe.src    = 'javascript:"";';
+        iframe.onload = function () {
+            var doc = iframe.contentDocument;
 
-    strictEqual(nativeMethods.querySelectorAll.call(document, '.' + SHADOW_UI_CLASSNAME.selfRemovingScript).length, 0);
+            // NOTE: Without setTimeout the error does not reproduced
```

Comment:
`// NOTE: Without wrapping in 'setTimeout' function the error is not reproduced`